ci/README.md

```diff
@@ -0,0 +1,67 @@
+#### What is functional test:
+
+Functional CI is about running functional tests in ci.centos.org for the whole build pipeline
+to check its end to end functionality. This aims to check behavior for all the stages in the 
+pipeline and matches with expected one. If there is any difference in expected and actual status
+of the stages, CI fails and prints all the logs for the pipeline run.
+
+#### Purpose:
+
+This is to test the end to end functionality of the service before deploying it to some infra. 
+CI aims to make sure for an early feedback loop and runs on each push to the repository.
+
+#### How it works:
+![Container Pipeline Functional CI Diagram](../docs/diagrams/functional_ci.png)
+
+[A job for Functional CI is setup in ci.centos.org](https://ci.centos.org/view/Container/job/centos-container-pipeline-service-ci-pr-openshift/)
+for running the ci tests on pull requests raised against [openshift branch of container-pipeline-service](https://github.com/CentOS/container-pipeline-service/tree/openshift).
+Once a PR is raised or updated gihub generates webhook which triggers a run of this job. 
+
+This jobs starts with following steps
+
+#### Get duffy nodes from [cico duffy pool](http://wiki.centos.org/QaWiki/CI/Duffy):
+
+* Gets 4 duffy nodes from duffy node pool
+  * One is used as ansible controller
+  * One as NFS node and Container-registry
+  * One as OpenShift master
+  * One as OpenShift node
+  
+If the job is not able to get nodes from CICO duffy CI is marked as FAILURE.
+
+#### Setting up nodes and deploying openshift cluster:
+
+Once nodes are avaialble, it sets up all the nodes according to their roles. 
+* `/etc/hosts entry` is done to all the nodes for resolving all the other nodes
+* `ansible-controller` nodes ssh pub key is added to root user of all the other nodes
+* Iptables allows `all the communicaitons among the nodes` and no node is exposed to out side
+* `openshift-ansible` RPM is used for setting up the cluster using `hosts.ci` as inventory file
+
+If openshift cluster and nodes are not setup to desired state, CI is  marked as FAILURE.
+
+#### Deploying Jenkins and pipeline service:
+
+* CI deploys the jenkins and pipeline service as normal deployment with following parameters
+    * `CONTAINER_INDEX` points to https://github.com/centos/container-index.git
+    * `CONTAINER_INDEX_BRANCH` points to `ci`
+
+#### Running CI checks:
+
+CI job runs in following sequence
+* `seed-job` runs first. This parses the container-index and creates the job-pipelines in openshift cluster
+* `ci-job for success check` runs second. This triggers a build for the `bamachrn-python-release` build config.
+Then matches all the stages with expected status.
+  * Waits for a stage to `complete or fail` then checks the status and marks the pipeline as `pass or fail` based on that
```

Comment:
Again, why does "complete or fail" have to be in codeblocks?

Reply:
Similarly for "pass or fail".

Reply:
I wanted to highlight it, changed it to italic